### ISSUE TITLE
Analyse performance improvement (Parramatta) dashboards

### ISSUE THREAD

**Issue by lexvand** [nens/parramatta-dashboard#222]
<p>Parramatta’s only complaint about their dashboard is the time it takes to load the (timeseries) tiles.</p>

<p>Determine what the actual bottleneck is, backend or frontend. And what would be possible measures to decrease loading times?</p>

<p>____________________________________________________________________________________________________________</p>

<p>Conclusion:<br/>
The majority of the delay (25 seconds up tomore then 60 seconds) seems to be in the Plotly engine.<br/>
I would like a second pair of eyes to confirm this findings</p>

<p>Analysis by Tom:<br/>
I inspected the network tab in both chrome and firefox but this was very confusing.<br/>
In chrome it seemed that the delay was because of the server.<br/>
In firefox it seemed that the first 20 seconds of delay was on server and last 20 second was on client<br/>
I therefore started commenting out code.</p>

<ul>
	<li>If I comment out the plotly then the complete app is loaded in under 15 seconds in both chrome as firefox.<br/>
Without unccommenting plotly this is at least 40 seconds and sometimes up to 1:30 minutes</li>
</ul>

<p>Options:</p>

<ul>
	<li>Pass plotly less data<br/>
Plotly is now passed a lot of data: 16 graphs, 4 lines each, 100 values per line (p[robably more)</li>
	<li>make plotly refresh less often<br/>
Each time new data arrives the entire dashboard is rerendered. Can we rerender lesss often or only rerender those updated tiles?<br/>
I looked intoo this for the infinite api call problems but couldnot improve this. It was partly because of how react and redux are working.</li>
	<li>Choose another technology then plotly ( which may have similar problems)</li>
	<li>show less data on the dashboard, devide over several dashboards</li>
</ul>

<p>___________________________________________________________________</p>

<p>CONCLUSION </p>

<p>after investigation together with Remco:</p>

<p>Cause:</p>

<p>The plotly library (or probably tiles in general) are retriggered to often.</p>

<p>We see that by commenting out plotly we reach a factor 4 improvement increase.</p>

<p>We should trigger Plotly less often to rerender.</p>

<p>This currently happens because each tile is connected to the whole store.</p>

<p>Possible solutions:</p>

<p>communication redux metadata only in container component (tile)<br/>
This component passes it as props to the chart.<br/>
Chart thus does not need to be connect te redux metadata<br/>
In short: <br/>
Container metadata, <br/>
chart real data.</p>

<p>In map state to props we need to filter only the data we need.<br/>
In shouldComponentUpdate we decide if the data is changed<br/>
___________________________________________________________________</p>

<i>[FRNT-240] created by joeri.verheijden</i>
